flink-end-to-end-tests/test-scripts/common_kubernetes.sh

```diff
@@ -50,17 +59,22 @@ function setup_kubernetes_for_linux {
 
     if ! [ -x "$(command -v minikube)" ]; then
       echo "Installing minikube $MINIKUBE_VERSION ..."
-      curl -Lo minikube https://storage.googleapis.com/minikube/releases/$MINIKUBE_VERSION/minikube-linux-$arch && \
-          chmod +x minikube && sudo mv minikube /usr/bin/minikube
+      download_minikube_url="https://storage.googleapis.com/minikube/releases/$MINIKUBE_VERSION/minikube-linux-$arch"
+      if ! retry_times ${RETRY_COUNT} ${RETRY_BACKOFF_TIME} "curl --fail -Lo minikube ${download_minikube_url}"; then
+        echo "ERROR: Could not download minikube. Aborting..."
+        exit 1
+      fi
+      chmod +x minikube && sudo mv minikube /usr/bin/minikube
     fi
 
     # conntrack is required for minikube 1.9 and later
     sudo apt-get install conntrack
     # crictl is required for cri-dockerd
-    local crictl_version crictl_archive
-    crictl_version="v1.24.2"
-    crictl_archive="crictl-$crictl_version-linux-${arch}.tar.gz"
-    wget -nv "https://github.com/kubernetes-sigs/cri-tools/releases/download/${crictl_version}/${crictl_archive}"
+    if ! retry_times ${RETRY_COUNT} ${RETRY_BACKOFF_TIME} download_circtl; then
+      echo "ERROR: Could not download crictl. Aborting..."
+      exit 1
+    fi
+
     sudo tar zxvf ${crictl_archive} -C /usr/local/bin
     rm -f ${crictl_archive}
```

Comment:
There are a few other artifact downloads below this line. Shall we create a generic download and retry function? We could align the tool for downloading (`curl` vs `wget`) there as well.